core/user/service.go

```diff
@@ -191,7 +187,12 @@ func (s Service) FetchCurrentUser(ctx context.Context) (User, error) {
 
 	fetchedUser, err := s.repository.GetByEmail(ctx, email)
 	if err != nil {
-		return User{}, err
+		switch err {
+		case ErrNotExist:
+			return User{}, fmt.Errorf("%s for email %s", ErrNotExist, email)
+		default:
+			return User{}, err
```

Comment:
Shouldn't the default return `ErrInvalidEmail` ?

Reply:
Then do you think we should keep `ErrInvalidEmail` that were removed?

Reply:
I think since we have already declared ErrMissingEmail/ErrNotExist that explain the error better we might not need it. The other error returned from the repository are things like queryErr or parseErr can't be represented by ErrInvalidEmail. I also think that ErrInvalidEmail is very general, or would fit more if we do email validation.

Reply:
What is the status code return by the API gor that error?

Reply:
There is no uniform return code in shield so it based on the implementation in handler. Some aren't even handling it. Guess this need to be fixed to now?

Reply:
In each handler we catch `ErrInvalidEmail` and return unauthenticated error
If we return `ErrNotExist` and not handling it in the handler, it will return 500 which is misleading
That is why I was suggesting to use or keep the error `ErrInvalidEmail` 
https://github.com/goto/shield/blob/81b8706093a49bd3236b6079b6b01b029c93919c/internal/api/v1beta1/org.go#L87

Reply:
For case in update handlers, will add block to catch the `ErrInvalidEmail`